projects/packages/forms/src/dashboard/inbox/bulk-actions-menu.js

```diff
@@ -1,14 +1,56 @@
-import { Button } from '@wordpress/components';
+import { AntiSpamIcon } from '@automattic/jetpack-components';
+import { Button, Spinner } from '@wordpress/components';
 import { useDispatch, useSelect } from '@wordpress/data';
+import { useCallback, useState } from '@wordpress/element';
 import { __ } from '@wordpress/i18n';
+import { config } from '../';
 import { doBulkAction } from '../data/responses';
 import { STORE_NAME } from '../state';
 import { ACTIONS, TABS } from './constants';
 
+/**
+ * Custom temporary handler for check-for-spam action based on grunion_check_for_spam.
+ *
+ * @param {number} offset - Offset for the query.
+ * @returns {Promise} Promise that resolves once checking for spam has finished.
+ */
+const checkForSpam = ( offset = 0 ) => {
+	const limit = 100;
+	const body = new FormData();
+
+	body.append( 'action', 'grunion_recheck_queue' );
+	body.append(
+		`jetpack_check_feedback_spam_${ config( 'blogId' ) }`,
+		config( 'checkForSpamNonce' )
+	);
+	body.append( 'offset', offset );
+	body.append( 'limit', limit );
+
+	return fetch( window.ajaxurl, { method: 'POST', body } )
```

Comment:
Is there a special reason we're not using apiFetch here?

Reply:
Hmm, not a special reason... if anything we're trying to avoid any magic 🤔